snapshot/lib/snapshot/runner.rb

```diff
@@ -70,6 +81,56 @@ def work
       end
     end
 
+    def patch_scheme_to_save_screenshots
+      scheme = Snapshot.config[:scheme]
+      scheme_absolute_path = File.join(File.expand_path(".", Snapshot.project.path), 'xcshareddata', 'xcschemes', scheme + ".xcscheme")
+
+      scheme_xml_doc = fetch_scheme_xml_file(scheme_file_path: scheme_absolute_path)
+
+      backup_scheme_file_path = scheme_absolute_path + "_original"
+      backup_scheme(scheme_absolute_path: scheme_absolute_path, backup_scheme_absolute_path: backup_scheme_file_path)
+
+      scheme_elements = scheme_xml_doc.elements['Scheme']
+      test_action = scheme_elements.elements['TestAction']
+      test_action.attributes["systemAttachmentLifetime"] = "keepAlways"
```

Comment:
Now we can use XPath to find the node, which is a more safe way of search and replace. This way we also don't duplicate the attribute if it already exists which might happen with a search and replace. Two benefits for the price of 1 🎉